Update trigger_step.md (#2887)
buildkite/docs: pages/pipelines/trigger_step.md

If neither condition is true, the build will fail, and builds on subsequent pipelines will not be triggered.

> If using bot users(unregistered users who are not part of any team) to trigger pipelines, make sure you have shared group which has build permission across parent and child pipeline.

Reviewer: Thanks for your contribution @PriyaSudip ! 🎉 

When you say 'group' do you mean a 'team' (as in Buildkite's _teams_ feature)?

```suggestion
If using bot users (unregistered users who are not part of any team) to trigger pipelines, make sure you have shared team which has the build permission on parent and child pipelines.
```

Author: Thanks @gilesgas yes , team would be an appropriate word!
